_posts/2021-09-13-user-defined-value-types.md

```diff
@@ -0,0 +1,151 @@
+---
+layout: post
+published: true
+title:  User Defined Value Types in Solidity
+date:   2021-09-13
+author: Solidity Team
+category: Explainers
+---
+
+Solidity [Solidity v0.8.8](https://github.com/ethereum/solidity/releases/tag/v0.8.8) introduces
+[user defined value
+types](https://docs.soliditylang.org/en/develop/types.html?#user-defined-value-types) as a means to
+create zero-cost abstractions over an elementary value type.
+
+## Motivation
+
+A problem with primitive value types is that they are not very descriptive. For example, one may
+want to use `uint128` to store the price of some object as well as the quantity available.
+Furthermore, having stricter type rules to avoid intermingling of the two different concepts are
+also quite useful. For example, assigning a quantity to a price or vice versa should be disallowed
+
+A [struct](https://docs.soliditylang.org/en/develop/types.html#structs) solves both of these
+problems, for example, by abstracting the price and the quantity as structs. The following solidity
+snippet demonstrates this:
```

Comment:
*Both* problems? Well, ok, you had ``Furthermore,...`` above, but to me it seems like one and the same problem... maybe

```suggestion
One option for solving this issue is using [structs](https://docs.soliditylang.org/en/develop/types.html#structs). For example, price and quantity can be abstracted as structs as follows:
```
?